ipld/resolver/src/behaviour/discovery.rs

```diff
@@ -153,39 +93,54 @@ pub struct Behaviour {
 }
 
 impl Behaviour {
-    /// Create a [`discovery::Behaviour`] from this configuration.
-    pub fn new(config: Config) -> Self {
-        let kademlia_opt = if config.enable_kademlia {
+    /// Create a [`discovery::Behaviour`] from the configuration.
+    pub fn new(nc: NetworkConfig, dc: Config) -> Result<Self, ConfigError> {
+        if nc.network_name.is_empty() {
+            return Err(ConfigError::InvalidNetwork(nc.network_name));
+        }
+        // Parse static addresses.
+        let mut static_addresses = Vec::new();
+        for multiaddr in dc.static_addresses {
+            let mut addr = multiaddr.clone();
+            if let Some(Protocol::P2p(mh)) = addr.pop() {
+                let peer_id = PeerId::from_multihash(mh).unwrap();
+                static_addresses.push((peer_id, addr))
+            } else {
+                return Err(ConfigError::InvalidBootstrapAddress(multiaddr));
+            }
+        }
+
+        let kademlia_opt = if dc.enable_kademlia {
             let mut kad_config = KademliaConfig::default();
-            let protocol_name = format!("/ipc/kad/{}/kad/1.0.0", config.network_name);
+            let protocol_name = format!("/ipc/kad/{}/kad/1.0.0", nc.network_name);
```

Comment:
What about removing a kad? :)
```suggestion
            let protocol_name = format!("/ipc/{}/kad/1.0.0", nc.network_name);
```

Reply:
OMG :see_no_evil: 

Comes from Forest but how can I be so blind :flushed: